### ISSUE TITLE
OJS categories include catalog prefix in the URL

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#4436]
The new browse-able categories in OJS include `catalog` in their URL.

```
http://localhost:8000/publicknowledge/catalog/category/test
```

This should be dropped so that categories come directly off of the context path:

```
http://localhost:8000/publicknowledge/category/test
```

The URL naming pattern should probably be plural, too, so that we can eventually have a sensible name for an index page that lists all categories:

```
http://localhost:8000/publicknowledge/categories
http://localhost:8000/publicknowledge/categories/test
```

There are [UX and SEO reasons](https://moz.com/learn/seo/url) for this, but they're pretty minor. Mostly, I think good URL structure just helps put forward a professional vibe, and we should change it before journals start publishing on it, since moving an indexed URL can result in them taking a hit on their SEO.

**Comment by asmecher** [comment]
Hi @NateWr -- I gave this a bit of thought and went with `catalog` both for consistency with OMP and to keep aligned with our (centuries-old) `page`/`op` URL scheme. We may also need more of the OMP `catalog` URL namespace for https://github.com/pkp/pkp-lib/issues/1407. In any case, I'm open to changing this, but nervous about introducing these new endpoints and then changing them in the next release.

**Comment by NateWr** [comment]
I hear ya. My worry is putting it out as `catalog` in 3.1.2 and then needing to modify it -- although if we had to, the gap between 3.1.2 and 3.2 would probably be the least disruptive.

I just think that `catalog` feels off for OJS, and if we're sharing code like this we should try to find a way to separate the actual `page`/`op` and the code underneath. Maybe just a parent class in each app that defines the specific handlers, but the base class delivers the goods? I realize this might take some untangling on the OMP side.

For continuous publishing (#1407), I still wouldn't envision a `/catalog` style page. In such a scenario, it seems more common to have the journal index be the base page, and from there to have different browsing options like `/latest`, `/categories/<category>`, `/sections/<section>`, `/search` and so on.

Also, for culanth, they'll probably want the URL to identify each category as a "collection". It should be possible, in the theme plugin, to hijack the request and route it to a custom `PageHandler` that simply wraps the existing handler with it's own `op`, right? We still, then, have to deal with how the URLs are generated in the theme, and return `404` for the `categories` op, but shouldn't be a simple customization to maintain over time.

On a separate note: since categories is in core, should browsing by section go into core too, rather than keeping it in a plugin?

**Comment by asmecher** [comment]
I don't mind deprecating a URL in a future release and adding a stub function in the handler to do a redirect -- but I'm nervous about editing the URL handling code for 3.1.2, so I'd suggest we come back to this post-3.1.2.

As for browse by section, yes, that might make sense to build in; I suspect it'll be nicer to themers to have a bunch of things to optionally theme in OJS rather than some themes that theme other plugins, which will be a lot more nebulous. Not to say that we shouldn't have plugins that extend the interface, but we shouldn't be afraid to build front-end options into OJS, IMO.

**Comment by NateWr** [comment]
:+1: I've created an issue for bringing browse by section to core (https://github.com/pkp/pkp-lib/issues/4498) and I can try to tackle some of the category stuff alongside that work.

**Comment by NateWr** [comment]
This should get addressed in #5932.

**Comment by NateWr** [comment]
Closing this in favor of #5932.
